c7n/filters/offhours.py

```diff
@@ -340,7 +343,7 @@ class OffHour(Time):
 
     schema = type_schema(
         'offhour', rinherit=Time.schema, required=['offhour', 'default_tz'],
-        offhour={'type': 'integer', 'minimum': 0, 'maximum': 24})
+        offhour={'type': 'integer', 'minimum': 0, 'maximum': 23})
```

Comment:
note, 24 isn't a valid value in the tag. If you put 24 in the tag value, you get this: custodian.offhours:WARNING Invalid schedule on resource:i-05badc67a1f80e525 value:off=(m-f,24);on=(m-f,8)

0-23 are the only valid values, 23 is 11pm-12am, 0 is 12am to 1am.